### PR TITLE
Check contstraints for memory parameters

### PR DESCRIPTION
**Pull request by mike1729**
Adds a simple test to check if parameters are right.

### CODE REVIEW

File: bin/runtime/src/lib.rs [hunk 1]
```diff
@@ -1015,10 +1015,35 @@ impl_runtime_apis! {
 
 #[cfg(test)]
 mod tests {
-    use crate::VERSION;
+    use super::*;
+    use primitives::HEAP_PAGES;
+    use smallvec::Array;
```
Comment (deuszx) [inline bin/runtime/src/lib.rs:1020]: Where do we use this exactly? IIRC, `smallvec` gives _some_ performance gains by pre-allocating space up to some small numbers of vec elements. I'm not sure it's worth adding new dependency just for it, especially in tests.

File: bin/runtime/src/lib.rs [hunk 2]
```diff
@@ -1015,10 +1015,35 @@ impl_runtime_apis! {
...
+        let max_heap_size = <Runtime as pallet_contracts::Config>::Schedule::get()
+            .limits
+            .max_memory_size();
+        // Max call depth is CallStack::size() + 1
+        let max_call_depth = <Runtime as pallet_contracts::Config>::CallStack::size() as u32 + 1;
+        // Max code len
+        let max_code_len: u32 = <Runtime as pallet_contracts::Config>::MaxCodeLen::get();
+
+        let lhs = max_call_depth * (72 * max_code_len + max_heap_size + MAX_STACK_SIZE);
+        let rhs = MAX_RUNTIME_MEM * 3 / 4;
```
Comment (DamianStraszak) [inline bin/runtime/src/lib.rs:1045]: Could you write a one-line comment where this 3/4 is coming from, and that this is not actually enforced in the runtime :P

**Review by DamianStraszak — APPROVED** [review]

File: bin/runtime/src/lib.rs (hunk 1 above)
Comment (DamianStraszak) [reply bin/runtime/src/lib.rs:1020]: We can maybe move to `dev-dependencies`?

Comment (mike1729) [reply bin/runtime/src/lib.rs:1020]: @deuszx `CallStack` is of this type so I need to import it to call `CallStack::size`.
@DamianStraszak sure.

File: bin/runtime/src/lib.rs (hunk 2 above)
Comment (deuszx) [reply bin/runtime/src/lib.rs:1045]: I was hesitating whether I should ask the same question or I will come across as 🙈 🙈 🙉

Comment (mike1729) [reply bin/runtime/src/lib.rs:1048]: Done.

**Pull request merged**